src/redux/slices/authSlice.js

```diff
@@ -1,21 +1,128 @@
 import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 import axios from 'axios';
 
+const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:3000';
+
+/**
+ * requestLogin accepts an object { email, password } which is used to
+ * log the user in.
+ *
+ * The returned token is placed in both localStorage and the redux slice.
+ */
+export const requestLogin = createAsyncThunk(
+  'auth/requestLogin',
+  async (credentials, { rejectWithValue }) => {
+    try {
+      const { email, password } = credentials;
+      let { data } = await axios.post(API_URL + '/api/auth/login', {
+        email,
+        password,
+      });
+
+      if (data.token) {
+        localStorage.setItem('token', data.token);
+        return data;
+      } else {
+        throw new Error('Failed token creation');
+      }
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
+/**
+ * tryToken takes no parameters, but it does pull token from localStorage
+ * Once verified, the user info and token are placed in slice
+ */
+export const tryToken = createAsyncThunk(
+  'tryToken',
+  async (placeholder, { rejectWithValue }) => {
+    try {
+      const token = localStorage.getItem('token');
+
+      if (!token) {
+        throw new Error('No token found in localStorage');
+      }
+
+      const { data } = await axios.get(API_URL + '/api/auth', {
+        headers: {
+          authorization: token,
+        },
+      });
+      if (data) {
+        return { data, token };
+      }
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
 const authSlice = createSlice({
   name: 'auth',
   initialState: {
     token: '',
-    auth: {},
+    user: {},
+    isLoading: false,
+    error: '',
   },
   reducers: {
-    testAuth: (state) => {
-      // delete me
-      state.token = 'test';
+    resetStatus: (state) => {
```

Comment:
Since we will likely have a resetStatus for each slice, can this be renamed to resetAuthStatus so when it is used, it is clear what it is doing.